modules/nw-ovn-k-anp-best-practices.adoc

```diff
@@ -0,0 +1,58 @@
+= Designing AdminNetworkPolicy
+
+When building `AdminNetworkPolicy` (ANP), you should consider the following when creating your policies:
+
+//This will need to be added in at 4.16.1
+
+//* You should create `AdminNetworkPolicy` at different priorities so that precedence is well defined. It is not guaranteed which policy will take precedence when overlapping ANP are created.
+* Administrators should build ANP that apply to user namespaces not system namespaces.
+
+[IMPORTANT]
+====
+Applying ANP and BANP to system namespaces is not supported because this can leave your cluster unresponsive and in a non-functional state.
+====
+
+* Since `0-100` is the supported priority range, it is always better when designing your ANP to use a middle range like `30-70` because this leaves some placeholder
+for priorities before and after. Even in the middle range, you might want to leave gaps so that as your infrastructure requirements evolve over time, you are able to insert new ANPs when needed at the right priority level. If you pack your ANPs, then you might need to recreate all of them to accommodate any changes in the future.
+
+* When using `0.0.0.0/0` or `::/0` to create a strong `Deny` policy, it is recommended that you check to make sure that you have higher priority `Allow` rules for essential traffic.
+
+* `Pass` is a special action used to delegate policy decision to the `NetworkPolicy` layer. You might consider using this action when considering the interoperability between ANP and your `NetworkPolicy`.
+
+* Selectors across multiple rules should not overlap. This ensures that that the same IPs do not appear in multiple policies, which can cause performance and scale limitations due to an OVN-Kubernetes limitation.
+
+* Using `namedPorts` is not performance efficient since each name has to get evaluated against the `subject` of the policy and `peer` pod's container ports. You might use this with caution in larger clusters where you have many pods and each pod has many containers.
+
+[id="considerations-using-banp_{context}"]
+== Considerations for using BaselineAdminNetworkPolicy
+
+* There is only one `BaselineAdminNetworkPolicy` (BANP) for the cluster be careful in defining it. The following are supported uses for BANP that administrators might consider in designing their BANP:
+
+** You can set a default deny policy for ingress in user namespaces. This BANP will make it so that users have to add `NetworkPolicy` to allow the ingress traffic that they want to allow, and if they do not add network policies for ingress it will be denied.
+
+** You can set a default deny policy for egress in user namespaces. This BANP will make it so that users have to add `NetworkPolicy` to allow the egress traffic that they want to allow, and if they do not add network policies it will be denied.
+
+** You can set a default allow policy for egress to DNS.This BANP will make it so that users in namespaces do not have to set an allow egress `NetworkPolicy` to DNS.
```

Comment:
(This is only interesting in combination with the previous "default-deny egress" policy; if you don't have a default-deny egress policy, you don't need an allow-egress-to-DNS policy because it will be allowed anyway.)